jupyter_collaboration/handlers.py

```diff
@@ -173,13 +173,20 @@ def __init__(
                         file_id,
                         file_format,
                         file_type,
-                        file_id_manager,
+                        self._file_id_manager,
                         self.contents_manager,
                         self.log,
                         self.settings["collaborative_file_poll_interval"],
                     )
                     self.files[file_id] = file
 
+                else:
+                    self._emit(
+                        "initialize",
+                        "There is another collaborative session accessing the same file.\nThe synchronization between rooms is not supported and you might lose some of your progress.",
```

Comment:
From [here](https://github.com/jupyterlab/jupyter_collaboration/pull/121#issuecomment-1503510439), I thought synchronization between rooms was supported. Could you clarify?

Reply:
I guess we should rather say that it is _experimental_.

Reply:
The UX of synchronization between rooms is pretty bad. With a good connection, a noticeable delay makes it unpredictable.

When two users are editing the same file using different models (for example, a user editing a notebook using the notebook view and another one using the plain text editor), propagating the changes from user A in room A to user B in room B takes a few seconds (there is a 1s delay to save the document + 1s delay for polling the file assuming that the communication and load delay is 0). During those 2s, user B can write a sentence that will disappear without any notification or asking the user if they want to overwrite the content.

In addition, we use the default content manager to load and save documents. When loading or saving a notebook, the content manager throws an error if the notebook is invalid (so it doesn't load/save the content). If user A in room A is editing the notebook using the plain text view, room B (with user B using the notebook editor) will be updated only when room A saves a valid notebook. Suppose user A forgets a curly bracket (which often happens because there is no visual indicator). In that case, everything user B is typing will be deleted once user A fixes the notebook without allowing user B to overwrite what is on the disk. Furthermore, if user B doesn't realize that their changes are not being saved, they can be typing for hours, and everything will be deleted.

The combination of auto-saving and room synchronization can be catastrophic; in my opinion, we can not call that "supported".